HabitTracker.AshtonLeeSeloka/DataService/DataServices.cs

```diff
@@ -0,0 +1,312 @@
+using Microsoft.Data.Sqlite;
+using System.Globalization;
+using DBItem;
+namespace DataService
+{
+	public class DataServices
+	{
+		public List<DBItems> dataRecords = new List<DBItems>();
+
+		/// <summary>
+		/// General methode used to recieve user input
+		/// </summary>
+		/// <param name="messageToUser">Message to write to console</param>
+		/// <returns>USer input as a string value</returns>
+		public string UserUnput(string messageToUser) 
+		{
+			Console.WriteLine(messageToUser);
+			return Console.ReadLine(); 
+		}
+
+		/// <summary>
+		/// Validates user date input and saves to variable if succesful
+		/// </summary>
+		/// <param name="messageToUser">Prompt displayed to user</param>
+		/// <returns>Correct date value or null</returns>
+		public string? DateInput(string messageToUser) 
+		{
+			Console.Write(messageToUser);
+			string? date = Console.ReadLine();
+
+			if (date == "0")
+				return null;
+
+			while (!DateTime.TryParseExact(date, "dd-MM-yyyy", new CultureInfo("en-Us"), DateTimeStyles.None, out _)) 
+			{
+				Console.Write("\n Invalid date. Enter Date with format of 'dd-MM-yy', or type '0 to exit'\n");
+				date = Console.ReadLine();
+
+				if (date == "0")
+					return null;
+			}
+			return date;	
+		}
+
+
+		/// <summary>
+		/// Creates an instance of the HabbitLogger Database if not present
+		/// </summary>
+		public void CreateDB() 
+		{
+			var connectionString = @"Data Source = HabbitLogger.db";
+
+			try
+			{
+				using (var connection = new SqliteConnection(connectionString))
+				{
+					connection.Open();
+					var tableCmd = connection.CreateCommand();
+					tableCmd.CommandText = @"CREATE TABLE IF NOT EXISTS habbit_logger (
+											Id INTEGER PRIMARY KEY AUTOINCREMENT,
+											Habit Text,
+											Date Text,
+											Quantity Integer						
+											)";
+					tableCmd.ExecuteNonQuery();
+				}
+			}
+			catch (Exception ex) 
+			{
+				Console.WriteLine(ex.Message);
+			}
+		}
+
+
+		/// <summary>
+		/// Inserts data into the table Database 
+		/// </summary>
+		public void InsertRecord() 
+		{
+			CreateDB();
```

Comment:
🟠 **Weird Method Placement**

💡 Why are you only creating the database on Insert? What happens if the user performs another action first, like View Records?

Reply:
I have moved this method call to the start of the program , ensuring that a db is always present to perform crud operations against.